python/plugins/processing/algs/qgis/Rasterize.py

```diff
@@ -0,0 +1,328 @@
+
+        self.addParameter(
+            # TODO use QgsProcessingParameterMapLayer when
+            # the LayerWidgetWrapper class will be implemented
+            QgsProcessingParameterRasterLayer(
+                self.LAYER,
+                description=self.tr(
+                    'Layer to render. Will only be used if the map theme '
+                    'is not set. '
+                    'If both, map theme and layer are not '
+                    'set, the current map content will be rendered.'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterExtent(self.EXTENT, description=self.tr(
+                'The minimum extent to render. Will internally be extended to '
+                'be a multiple of the tile sizes.')))
+        self.addParameter(
+            QgsProcessingParameterNumber(
+                self.TILE_SIZE,
+                self.tr('Tile size'),
+                defaultValue=1024))
+
+        self.addParameter(QgsProcessingParameterNumber(
+            self.MAP_UNITS_PER_PIXEL,
+            self.tr(
+                'Map units per '
+                'pixel'),
+            defaultValue=100,
+            minValue=0,
+            type=QgsProcessingParameterNumber.Double
+        ))
+
+        # We add a raster layer as output
+        self.addParameter(QgsProcessingParameterRasterDestination(
+            self.OUTPUT,
+            self.tr(
+                'Output layer')))
+
+    def name(self):
+        # Unique (non-user visible) name of algorithm
+        return 'Rasterize'
+
+    def displayName(self):
+        # The name that the user will see in the toolbox
+        return self.tr('Convert map to raster')
+
+    def group(self):
+        return self.tr('Raster tools')
+
+    def tags(self):
+        return self.tr('layer,raster,convert,file,map themes,tiles').split(',')
+
+    # def processAlgorithm(self, progress):
+    def processAlgorithm(self, parameters, context, feedback):
+        """Here is where the processing itself takes place."""
+
+        # The first thing to do is retrieve the values of the parameters
+        # entered by the user
+        map_theme = self.parameterAsString(
+            parameters,
+            self.MAP_THEME,
+            context)
+
+        layer = self.parameterAsLayer(
+            parameters,
+            self.LAYER,
+            context)
+
+        extent = self.parameterAsExtent(
+            parameters,
+            self.EXTENT,
+            context)
+
+        tile_size = self.parameterAsInt(
+            parameters,
+            self.TILE_SIZE,
+            context)
+
+        mupp = self.parameterAsDouble(
+            parameters,
+            self.MAP_UNITS_PER_PIXEL,
+            context)
+
+        output_layer = self.parameterAsOutputLayer(
+            parameters,
+            self.OUTPUT,
+            context)
+
+        # This probably affects the whole system but it's a lot nicer
+        osgeo.gdal.UseExceptions()
```

Comment:
@marioba can you take that out? I fear the side effects of this, so I'd prefer to see this in another pull request where it's set in `qgis/__init__.py` next to the `sip.setapi()` call and check what effects it has on the whole system.